core/Common.php

```diff
@@ -256,6 +256,23 @@ public static function mb_strtoupper($string)
         return $string;
     }
 
+    /**
+     * Secure wrapper for unserialize, which by default disallows unserializing classes
+     *
+     * @param string $string String to unserrialize
+     * @param array $allowedClasses Class names that should be allowed to unserialize
+     *
+     * @return mixed
+     */
+    public static function safe_unserialize($string, $allowedClasses = [])
+    {
+        if (PHP_MAJOR_VERSION >= 7) {
+            return @unserialize($string, ['allowed_classes' => empty($allowedClasses) ? false : $allowedClasses]);
+        }
+
+        return @unserialize($string);
+    }
```

Comment:
Should unserialize errors be reported in any way? Is it a problem if an exception is thrown here?

Reply:
I see a lot of uses of `@unserialize()`, what about allowing callers to specify whether `Common::safe_unserialize()` should have errors silenced? Would that make things better?

Reply:
@diosmosis  you mean something like a third parameter that defines whether to ignore error or not?
Btw. unserialize doesn't throw an exception if the values can't be deserialized. It triggers a `E_NOTICE`. For PHP 7 it might be fine as it's "catchable" with the `Error` class. For PHP 5.x that simply results in errors depending on the php configuration.
Do you think it might be suitable to always ignore errors ion PHP 5 and always catch errors on PHP 7, but trigger a `Log::debug` in that case?

Reply:
I thought `ErrorHandler` might turn the notice into an exception, but I can't really remember and haven't tested it... But for this I figured we could let the callers silence `Common::safe_unserialize()` if they wanted, eg:

```
$value = @Common::safe_unserialize($value);
```

I thought about exceptions but noticed some of the uses you changed assume it will only return false on failure, so I guess we can't throw everywhere.

Reply:
👍 for logging, though

Reply:
added a debug log for that case and also added some tests...